lib/mayaUsd/fileio/transformWriter.cpp

```diff
@@ -98,12 +99,17 @@ void UsdMayaTransformWriter::_ComputeXformOps(
             continue;
         }
 
-        GfVec3d value = animChannel.defValue;
-        bool    hasAnimated = false;
-        bool    hasStatic = false;
+        GfVec3d    value = animChannel.defValue;
+        GfMatrix4d matrix = animChannel.defMatrix;
+        bool       hasAnimated = false;
+        bool       hasStatic = false;
         for (unsigned int i = 0u; i < 3u; ++i) {
             if (animChannel.sampleType[i] == _SampleType::Animated) {
-                value[i] = animChannel.plug[i].asDouble();
+                if (animChannel.isMatrix) {
+                    matrix = animChannel.GetSourceData(i).Get<GfMatrix4d>();
```

Comment:
I was wary of leaving uninitialized values. Normally they should never be accessed by code elsewhere, but I did not want them to be left with random values just in case. I can exit early as normally, they won't be accessed elsewhere.

Reply:
I might add default initialization to _AnimChannel to avoid uninitialized values.

Reply:
Ah, I was thinking of the loop at line 220, here there are noe value to be left uninitialized.